gax-java/gax/src/test/java/com/google/api/gax/AutoValuesWithDeprecatedThreetenFieldsTest.java

```diff
@@ -0,0 +1,288 @@
+package com.google.api.gax;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.google.api.gax.util.ThreetenFieldUpgrade;
+import com.google.api.gax.util.ThreetenFieldUpgrade.FieldRole;
+import com.google.api.gax.util.ThreetenTestAutoValueOverrideClass;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.logging.Logger;
+import javax.annotation.Generated;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.reflections.Reflections;
+
+@RunWith(JUnit4.class)
+public class AutoValuesWithDeprecatedThreetenFieldsTest {
+
+  private static final Logger logger = Logger.getLogger("");
+
+  @Test
+  public void testAutoValueClassesWithThreetenMethods()
+      throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+    List<Class<?>> targetClasses = getAutoValueClassesWithThreetenMethods();
```

Comment:
This is a smart way to test all autovalue classes, but I think we should test each of them individually. Because
1. Better readability and maintainability. We might be writing less code this way, but it's much harder for other developers to understand and maintain this test. 
2. No new public classes.
3. I don't think all test cases are need here. For example, testing `AutoValue getters should have no parameters` is like testing `AutoValue` instead of our feature.
That being said, I do see the confusion of how to test this project in general. So maybe we can create a one-pager testing guide to specify what level of testing is expected @lqiu96.

Reply:
Sounds good. There are a couple of general purpose test helper methods I could use instead of this scan approach